library/spdm_common_lib/libspdm_com_crypto_service.c

```diff
@@ -953,6 +953,8 @@ bool libspdm_verify_challenge_auth_signature(libspdm_context_t *spdm_context,
 
 #if LIBSPDM_RECORD_TRANSCRIPT_DATA_SUPPORT
     result = libspdm_calculate_m1m2(spdm_context, !is_requester, &m1m2);
+    m1m2_buffer = libspdm_get_managed_buffer(&m1m2);
+    m1m2_buffer_size = libspdm_get_managed_buffer_size(&m1m2);
 #else
     m1m2_hash_size = sizeof(m1m2_hash);
     result = libspdm_calculate_m1m2_hash(spdm_context, !is_requester, &m1m2_hash_size, &m1m2_hash);
```

Comment:
It would be more reasonable to move result check at line 969-971 to the line before 959 (check result before reset message_b/c/mut_b/mut_c).

Reply:
agree

Reply:
I see. Thanks for the explanation. We are all good here.